actions/publish-github-pages/entrypoint.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+# entrypoint.py
+#
+# #### PARAMETERS ####
+# Input parameters:
+# GITHUB_PAGES_DIRECTORY - the directory to write the rendered website files to
+# GITHUB_PAGES_REPO - repo containing documentation to be deployed to GitHub Pages
+# GITHUB_PAGES_PULL_BRANCH - the branch of the repo which contains the documentation
+# GITHUB_PAGES_PUSH_BRANCH - the branch of the repo which GitHub Pages will deploy to
+# CONTENT - comma-separated list of directories in the top level of the repo that contains documentation
+# HUGO_URL - URL of the Hugo site after deployment
+# HUGO_PUBLISH_DIR - directory in the repo that GitHub Pages will deploy to
+# HUGO_REPO - CloudPosse repository containing Hugo infrastructure
+# HUGO_CONFIG - location of to-be-written Hugo config file (actual location not important)
+# HTMLTEST_CONFIG - location of to-be-written htmltest config file (actual location not important)
+#
+# Example parameter values:
+# GITHUB_PAGES_DIRECTORY=github_pages
+# GITHUB_PAGES_REPO=https://github.com/cloudposse/docs # or, e.g., github.com/project/infrastructure.git
+# GITHUB_PAGES_PULL_BRANCH=master # or current-docs-feature-branch
+# GITHUB_PAGES_PUSH_BRANCH=production # or, e.g., docs
+# CONTENT=docs,content
+# HUGO_URL=cloudposse.github.io/docs
+# HUGO_PUBLISH_DIR=public
+# HUGO_REPO=https://github.com/cloudposse/docs
+# HUGO_CONFIG=hugo.config.new
+# HTMLTEST_CONFIG=.htmltest.config.new
+
+import os
+import re
+import subprocess
+from git import Repo
+from shutil import copy2, copytree, rmtree
+
+def main():
+    read_in_env_vars()
+    checkout_repos()
+
+    # Create a separate build folder, ${STAGING_DIR}, and populate it with the essential files from HUGO_REPO
+    # (The rest of this script assumes HUGO_REPO=https://github.com/cloudposse/docs.)
```

Comment:
Why don't we just make `main` call the functions. So lets move all this here and below to a new function.